pydantic_ai_slim/pydantic_ai/models/xai.py

```diff
@@ -0,0 +1,868 @@
+"""xAI model implementation using xAI SDK."""
+
+from collections.abc import AsyncIterator, Iterator, Sequence
+from contextlib import asynccontextmanager
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Literal, cast
+
+from typing_extensions import assert_never
+
+try:
+    import xai_sdk.chat as chat_types
+
+    # Import xai_sdk components
+    from xai_sdk import AsyncClient
+    from xai_sdk.chat import assistant, file, image, system, tool, tool_result, user
+    from xai_sdk.tools import code_execution, get_tool_call_type, mcp, web_search  # x_search not yet supported
+except ImportError as _import_error:
+    raise ImportError(
+        'Please install `xai-sdk` to use the xAI model, '
+        'you can use the `xai` optional group — `pip install "pydantic-ai-slim[xai]"`'
+    ) from _import_error
+
+from .. import _utils
+from .._run_context import RunContext
+from ..builtin_tools import CodeExecutionTool, MCPServerTool, WebSearchTool
+from ..exceptions import UnexpectedModelBehavior, UserError
+from ..messages import (
+    AudioUrl,
+    BinaryContent,
+    BuiltinToolCallPart,
+    BuiltinToolReturnPart,
+    CachePoint,
+    DocumentUrl,
+    FilePart,
+    FinishReason,
+    ImageUrl,
+    ModelMessage,
+    ModelRequest,
+    ModelRequestPart,
+    ModelResponse,
+    ModelResponsePart,
+    ModelResponseStreamEvent,
+    RetryPromptPart,
+    SystemPromptPart,
+    TextPart,
+    ThinkingPart,
+    ToolCallPart,
+    ToolReturnPart,
+    UserPromptPart,
+    VideoUrl,
+)
+from ..models import (
+    Model,
+    ModelRequestParameters,
+    StreamedResponse,
+    check_allow_model_requests,
+    download_item,
+)
+from ..profiles import ModelProfileSpec
+from ..providers import Provider, infer_provider
+from ..providers.grok import GrokModelName
+from ..settings import ModelSettings
+from ..usage import RequestUsage
+
+# Type alias for consistency
+XaiModelName = GrokModelName
+
+
+class XaiModelSettings(ModelSettings, total=False):
+    """Settings specific to xAI models.
+
+    See [xAI SDK documentation](https://docs.x.ai/docs) for more details on these parameters.
+    """
+
+    logprobs: bool
```

Comment:
All provider-specific model settings should be prefixed with `<provider>_`. Please add it, as well as the comment you can see on other model settings classes:

https://github.com/pydantic/pydantic-ai/blob/085e21f5ee3e5595020e300923fbb4f020862067/pydantic_ai_slim/pydantic_ai/models/groq.py#L117